instrumentation.go

```diff
@@ -122,6 +129,11 @@ func (c instConfig) applyEnv() instConfig {
 	if v, ok := os.LookupEnv(envTargetExeKey); ok {
 		c.target = &process.TargetArgs{ExePath: v}
 	}
+	if v, ok := os.LookupEnv(envServiceNameKey); ok {
```

Comment:
The service name can also be defined in [`OTEL_RESOURCE_ATTRIBUTES`](https://github.com/open-telemetry/opentelemetry-specification/blob/5f9bc69cebd2a45aa889d0a69ea9943b4257d85a/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable). It needs to be parsed after this to determine if it contains a service name.